### ISSUE TITLE
bigdecimal - Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

### ISSUE THREAD

**Issue by col-panic** [docker-library/redmine#327]
Using `redmine:5.1` and adding the plugin https://github.com/kontron/redmine_oauth to
the `plugins` directory, the errors added at the bottom are shown on startup.

After some research I found that some development tools are missing for inclusion of this library.
Unfortunately I could not find information on how to proper modify the image to allow for
inclusion of such plugins.

Any hel appreciated!

```
redmine-1  | You should probably keep only one of them.
redmine-1  | Remove any duplicate entries and specify the gem only once.
redmine-1  | While it's not a problem now, it could cause errors if you change the version of one of them later.
redmine-1  | Bundler can't satisfy your Gemfile's dependencies.
redmine-1  | Install missing gems with `bundle install`.
redmine-1  | Your Gemfile lists the gem puma (>= 0) more than once.
redmine-1  | You should probably keep only one of them.
redmine-1  | Remove any duplicate entries and specify the gem only once.
redmine-1  | While it's not a problem now, it could cause errors if you change the version of one of them later.
redmine-1  | Fetching gem metadata from https://rubygems.org/.........
redmine-1  | Resolving dependencies...
redmine-1  | Fetching bigdecimal 3.1.7
redmine-1  | Fetching base64 0.2.0
redmine-1  | Fetching uri 0.13.0
redmine-1  | Fetching hashie 5.0.0
redmine-1  | Installing base64 0.2.0
redmine-1  | Installing uri 0.13.0
redmine-1  | Installing hashie 5.0.0
redmine-1  | Fetching version_gem 1.1.4
redmine-1  | Installing bigdecimal 3.1.7 with native extensions
redmine-1  | Fetching jwt 2.8.1
redmine-1  | Installing version_gem 1.1.4
redmine-1  | Fetching faraday-net_http 3.1.0
redmine-1  | Fetching snaky_hash 2.0.1
redmine-1  | Installing jwt 2.8.1
redmine-1  | Installing snaky_hash 2.0.1
redmine-1  | Installing faraday-net_http 3.1.0
redmine-1  | Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
redmine-1  | 
redmine-1  |     current directory: /usr/local/bundle/gems/bigdecimal-3.1.7/ext/bigdecimal
redmine-1  | /usr/local/bin/ruby extconf.rb
redmine-1  | checking for __builtin_clz()... *** extconf.rb failed ***
redmine-1  | Could not create Makefile due to some reason, probably lack of necessary
redmine-1  | libraries and/or headers.  Check the mkmf.log file for more details.  You may
redmine-1  | need configuration options.
redmine-1  | 
redmine-1  | Provided configuration options:
redmine-1  | 	--with-opt-dir
redmine-1  | 	--without-opt-dir
redmine-1  | 	--with-opt-include
redmine-1  | 	--without-opt-include=${opt-dir}/include
redmine-1  | 	--with-opt-lib
redmine-1  | 	--without-opt-lib=${opt-dir}/lib
redmine-1  | 	--with-make-prog
redmine-1  | 	--without-make-prog
redmine-1  | 	--srcdir=.
redmine-1  | 	--curdir
redmine-1  | 	--ruby=/usr/local/bin/$(RUBY_BASE_NAME)
redmine-1  | /usr/local/lib/ruby/3.2.0/mkmf.rb:490:in `try_do': The compiler failed to
redmine-1  | generate an executable file. (RuntimeError)
redmine-1  | You have to install development tools first.
redmine-1  | 
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:616:in `block in try_compile'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:565:in `with_werror'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:616:in `try_compile'
redmine-1  | 	from extconf.rb:6:in `block in have_builtin_func'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:989:in `block in checking_for'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:354:in `block (2 levels) in postpone'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:324:in `open'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:354:in `block in postpone'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:324:in `open'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:350:in `postpone'
redmine-1  | 	from /usr/local/lib/ruby/3.2.0/mkmf.rb:988:in `checking_for'
redmine-1  | 	from extconf.rb:5:in `have_builtin_func'
redmine-1  | 	from extconf.rb:18:in `<main>'
redmine-1  | 
redmine-1  | To see why this extension failed to compile, please check the mkmf.log which can
redmine-1  | be found here:
redmine-1  | 
redmine-1  |   /usr/local/bundle/extensions/x86_64-linux/3.2.0/bigdecimal-3.1.7/mkmf.log
redmine-1  | 
redmine-1  | extconf failed, exit code 1
redmine-1  | 
redmine-1  | Gem files will remain installed in /usr/local/bundle/gems/bigdecimal-3.1.7 for
redmine-1  | inspection.
redmine-1  | Results logged to
redmine-1  | /usr/local/bundle/extensions/x86_64-linux/3.2.0/bigdecimal-3.1.7/gem_make.out
redmine-1  | 
redmine-1  |   /usr/local/lib/ruby/3.2.0/rubygems/ext/builder.rb:119:in `run'
redmine-1  |   /usr/local/lib/ruby/3.2.0/rubygems/ext/ext_conf_builder.rb:28:in `build'
redmine-1  |   /usr/local/lib/ruby/3.2.0/rubygems/ext/builder.rb:187:in `build_extension'
redmine-1  | /usr/local/lib/ruby/3.2.0/rubygems/ext/builder.rb:221:in `block in
redmine-1  | build_extensions'
redmine-1  |   /usr/local/lib/ruby/3.2.0/rubygems/ext/builder.rb:218:in `each'
redmine-1  |   /usr/local/lib/ruby/3.2.0/rubygems/ext/builder.rb:218:in `build_extensions'
redmine-1  |   /usr/local/lib/ruby/3.2.0/rubygems/installer.rb:846:in `build_extensions'
redmine-1  | /usr/local/lib/ruby/3.2.0/bundler/rubygems_gem_installer.rb:72:in
redmine-1  | `build_extensions'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/rubygems_gem_installer.rb:28:in `install'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/source/rubygems.rb:202:in `install'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/installer/gem_installer.rb:54:in `install'
redmine-1  | /usr/local/lib/ruby/3.2.0/bundler/installer/gem_installer.rb:16:in
redmine-1  | `install_from_spec'
redmine-1  | /usr/local/lib/ruby/3.2.0/bundler/installer/parallel_installer.rb:156:in
redmine-1  | `do_install'
redmine-1  | /usr/local/lib/ruby/3.2.0/bundler/installer/parallel_installer.rb:147:in
redmine-1  | `block in worker_pool'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/worker.rb:62:in `apply_func'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/worker.rb:57:in `block in process_queue'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/worker.rb:54:in `loop'
redmine-1  |   /usr/local/lib/ruby/3.2.0/bundler/worker.rb:54:in `process_queue'
redmine-1  | /usr/local/lib/ruby/3.2.0/bundler/worker.rb:90:in `block (2 levels) in
redmine-1  | create_threads'
redmine-1  | 
redmine-1  | An error occurred while installing bigdecimal (3.1.7), and Bundler cannot
redmine-1  | continue.
redmine-1  | 
redmine-1  | In Gemfile:
redmine-1  |   oauth2 was resolved to 2.0.9, which depends on
redmine-1  |     multi_xml was resolved to 0.7.1, which depends on
redmine-1  |       bigdecimal
redmine-1 exited with code 0
```
